sympy/solvers/ode/hypergeometric.py

```diff
@@ -0,0 +1,249 @@
+r'''
+This File contains all the matching and solution method for hypergeometric solver. Main reference
+for this algorithm is taken from  "Non-Liouvillian solutions for second order linear ODEs" by
+L. Chan, E.S. Cheb-Terrab. Although the paper describes an algorithm that should find all solutions
+with 2F1, 0F1 and 1F1,but Sympy only has 2F1 right now.
+The code could be extended to handle 0F1 and 1F1 as described in the paper.
+'''
+
+from sympy.core import S, Pow
+from sympy.core.function import expand
+from sympy.core.relational import Eq
+from sympy.core.symbol import Symbol, Wild
+from sympy.functions import exp, sqrt, hyper
+from sympy.integrals import Integral
+from sympy.polys import roots, gcd
+from sympy.polys.polytools import cancel, factor
+from sympy.simplify import collect, simplify, logcombine
+from sympy.simplify.powsimp import powdenest
+from sympy.solvers.ode.ode import get_numbered_constants
+
+
+def match_2nd_hypergeometric(eq, func):
+    x = func.args[0]
+    df = func.diff(x)
+    a3 = Wild('a3', exclude=[func, func.diff(x), func.diff(x, 2)])
+    b3 = Wild('b3', exclude=[func, func.diff(x), func.diff(x, 2)])
+    c3 = Wild('c3', exclude=[func, func.diff(x), func.diff(x, 2)])
+    deq = a3*(func.diff(x, 2)) + b3*df + c3*func
+    r = collect(eq,
+        [func.diff(x, 2), func.diff(x), func]).match(deq)
+    if r:
+        if not all([r[key].is_polynomial() for key in r]):
+            n, d = eq.as_numer_denom()
+            eq = expand(n)
+            r = collect(eq, [func.diff(x, 2), func.diff(x), func]).match(deq)
+
+    if r and r[a3]!=0:
+        A = cancel(r[b3]/r[a3])
+        B = cancel(r[c3]/r[a3])
+        return [A, B]
+    else:
+        return []
+
+def equivalence_hypergeometric(A, B, func):
```

Comment:
There should be two spaces here and also in other places. Currently the spacing is inconsistent in this file.

Reply:
I have fixed this in the latest commit of this PR, But I don't know why it is not showing this review 'outdated'.🤔